libpf/pfelf/testdata/Makefile

```
@@ -44,7 +44,6 @@ kernel-image: test.c
 ubuntu-kernel-image: test.c
 	$(CC) $< -s -o $@ -DLINUX_VERSION="\"Linux version 1.2.3 (Ubuntu 4.5.6)\\n\""
 
-# A fake go binary (with a .gopclntab section)
-go-binary: without-debug-syms
-	$(OBJCOPY) --add-section .gopclntab=/dev/null $< $@
+go-binary: gotest.go
+	go build -o go-binary -ldflags "-w -s" gotest.go
```

There should be a GOTOOLCHAIN directive, otherwise `TestGoVersion` will fail if the opentelemetry-ebpf-profiler-dev docker image is updated with a new Go version or one uses a different Go patch version locally.
```suggestion
	GOTOOLCHAIN=go1.23.6 go build -o go-binary -ldflags "-w -s" gotest.go
```

---

> otherwise `TestGoVersion` will fail

It is using `assert.GreaterOrEqual` so newer go versions get accepted also.

---

Yeah I tested with Go 1.23 and 1.24, should be something folks can ignore and will just work as the version marches forward.

---

As https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/409 differentiates on the Go version patch level (see interpreter/golabels/runtime_data.go
), shouldn't the test be precise as well and make sure the extracted patch version matches the expected patch version?

---

The version is returned by Go function that has its own test suite, so I would not be worried about this. The intent here is to test that the new method correctly ends up calling `buildinfo.Read` - not to test that it tests thing correctly.

I could argue that since the test executable is built with `make` and `go test` runs the test suite, there is case that I do `make`, then update go, and finally `go test` and I would expect this to work.

Because of this, the current `GreaterOrEqual` makes sense to me. I would be even happy if the test was just parsing the version with `version.IsValid`, or if the `GreaterOrEqual` was testing against `go1.10` or something ancient. That would more strongly hint on the purpose of the test.

---

It could I guess. Do we care more about the test showing that it gets the exact version string for a particular go version or do we care more about the test showing that it gets some version string that works with any go version going forward. I think the later is better. We could do both but the new go-label kernel tests already cover that version 1.23 and 1.24 are detected properly.

---

If we want equals test, perhaps the test suite can as additional test open `/proc/self/exe` and compare its version against `runtime.Version()` ?